src/cares_wrap.cc

```diff
@@ -2011,18 +2012,18 @@ void GetNameInfo(const FunctionCallbackInfo<Value>& args) {
   CHECK(uv_ip4_addr(*ip, port, reinterpret_cast<sockaddr_in*>(&addr)) == 0 ||
         uv_ip6_addr(*ip, port, reinterpret_cast<sockaddr_in6*>(&addr)) == 0);
 
-  GetNameInfoReqWrap* req_wrap = new GetNameInfoReqWrap(env, req_wrap_obj);
+  auto req_wrap = std::make_unique<GetNameInfoReqWrap>(env, req_wrap_obj);
 
   TRACE_EVENT_NESTABLE_ASYNC_BEGIN2(
-      TRACING_CATEGORY_NODE2(dns, native), "lookupService", req_wrap,
+      TRACING_CATEGORY_NODE2(dns, native), "lookupService", req_wrap.get(),
       "ip", TRACE_STR_COPY(*ip), "port", port);
 
   int err = req_wrap->Dispatch(uv_getnameinfo,
                                AfterGetNameInfo,
                                reinterpret_cast<struct sockaddr*>(&addr),
                                NI_NAMEREQD);
-  if (err)
-    delete req_wrap;
+  if (err == 0)
+    req_wrap.release();
```

Comment:
Please use `USE()` and add a comment that this indicates passing ownership to `req`

Reply:
Sorry, I missed this one. I'll update shortly. Thanks